internal/services/mssql/mssql_server_resource.go

```diff
@@ -266,214 +265,267 @@ func resourceMsSqlServerCreate(d *pluginsdk.ResourceData, meta interface{}) erro
-	}
+		// if you pass the Key ID you must also define the PrimaryUserAssignedIdentityID...
+		if payload.Properties.KeyId != nil && payload.Properties.PrimaryUserAssignedIdentityId == nil {
+			return fmt.Errorf("the `primary_user_assigned_identity_id` field must be specified to use the 'transparent_data_encryption_key_vault_key_id' in %s", id)
+		}
 
-	if v := d.Get("public_network_access_enabled"); !v.(bool) {
-		props.ServerProperties.PublicNetworkAccess = sql.ServerNetworkAccessFlagDisabled
-	}
+		payload.Properties.PublicNetworkAccess = pointer.To(servers.ServerPublicNetworkAccessFlagDisabled)
+		payload.Properties.RestrictOutboundNetworkAccess = pointer.To(servers.ServerNetworkAccessFlagDisabled)
 
-	if v := d.Get("outbound_network_restriction_enabled"); v.(bool) {
-		props.ServerProperties.RestrictOutboundNetworkAccess = sql.ServerNetworkAccessFlagEnabled
-	}
+		if v := d.Get("public_network_access_enabled"); v.(bool) {
+			payload.Properties.PublicNetworkAccess = pointer.To(servers.ServerPublicNetworkAccessFlagEnabled)
+		}
 
-	if d.HasChange("administrator_login_password") {
-		adminPassword := d.Get("administrator_login_password").(string)
-		props.ServerProperties.AdministratorLoginPassword = utils.String(adminPassword)
-	}
+		if v := d.Get("outbound_network_restriction_enabled"); v.(bool) {
+			payload.Properties.RestrictOutboundNetworkAccess = pointer.To(servers.ServerNetworkAccessFlagEnabled)
+		}
 
-	if v := d.Get("minimum_tls_version"); v.(string) != "Disabled" {
-		props.ServerProperties.MinimalTLSVersion = utils.String(v.(string))
-	}
+		if d.HasChange("administrator_login_password") {
+			adminPassword := d.Get("administrator_login_password").(string)
+			payload.Properties.AdministratorLoginPassword = pointer.To(adminPassword)
+		}
 
-	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, props)
-	if err != nil {
-		return fmt.Errorf("issuing update request for %s: %+v", id.String(), err)
-	}
+		if d.HasChange("minimum_tls_version") {
+			payload.Properties.MinimalTlsVersion = pointer.To(d.Get("minimum_tls_version").(string))
+		}
 
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		if response.WasConflict(future.Response()) {
-			return fmt.Errorf("SQL Server names need to be globally unique and %q is already in use", id.Name)
+		future, err := client.CreateOrUpdate(ctx, *id, payload)
+		if err != nil {
+			return fmt.Errorf("issuing update request for %s: %+v", id, err)
 		}
 
-		return fmt.Errorf("waiting for update of %s: %+v", id.String(), err)
-	}
+		if err = future.Poller.PollUntilDone(ctx); err != nil {
+			if response.WasConflict(future.HttpResponse) {
+				return fmt.Errorf("SQL Server names need to be globally unique and %q is already in use", id.ServerName)
+			}
 
-	d.SetId(id.ID())
+			return fmt.Errorf("waiting for update of %s: %+v", pointer.From(id), err)
+		}
```

Comment:
since this is an Update, the SQL Server name will already be reserved, so I think we should be able to reduce this too:

```suggestion
 		future, err := client.CreateOrUpdateThenPoll(ctx, *id, payload)
 		if err != nil {
 			return fmt.Errorf("issuing update request for %s: %+v", id, err)
 		}
```